Waidhoferj/CSC-308-309-Project
# added persistent auth

Fixes #119,
Fixes #127,
Fixes #75 
Fixes #86 

Implemented [Netlify's `gotrue-js`](https://github.com/netlify/gotrue-js) authentication library for the project. 

### What it does
- Fixes our problem with persisting user after reload.
- Easy to use wrapper around oAuth
- Allows us to sign up, log in, deactivate and reactivate users.

### What it doesn't do (yet)
- Provide security for our server. We have to reach out from the server and request confirmation of a the json cookie from the Netlify service. This could get complicated, so if we're serious about it, it can be addressed in another issue.
- I haven't checked for duplicate email errors and such.

## Testing
- boot up the servers
- verify that visiting any other page but sign up and login is impossible (using url)
- go to the `/sign-up` page.
- Sign up for an account
- Verify that it takes you to the map page afterward
- Click the log out button from the user profile page
- go to the login page
- log in
- verify that you got back in
